Assets/Scripts/Combat/Combantants/Combatant.cs

```diff
@@ -69,12 +81,29 @@ public void StartTurn()
 
     protected void TakeDamage(int damage)
     {
-        CurrentHealthPoints -= damage;
+        if (CurrentShieldHealth > 0)
+        {
+            CurrentShieldHealth -= damage;
+            if (CurrentShieldHealth <= 0) {
+                Shield.SetActive(false);
+                int leftoverDmg = -1 * CurrentShieldHealth;
+                CurrentHealthPoints -= leftoverDmg;
+            }
+        }
+
+        else
+            CurrentHealthPoints -= damage;
         
         if (CurrentHealthPoints <= 0)
             IsAlive = false;
     }
 
+    public void ApplyShield(int shieldHealth)
```

Comment:
Move the shield to the target. Also, adjust size to fit more snugly?

Reply:
This is done in the MagicShield Script, but I'll move it into the Combantant script since it makes more sense.